pkg/operator/ceph/config/defaults.go

```diff
@@ -65,15 +55,20 @@ func defaultFlagConfigs(fsid, mountedKeyringPath string) *Config {
 // DefaultCentralizedConfigs returns the default configuration options Rook will set in Ceph's
 // centralized config store. If the version of Ceph does not support the centralized config store,
 // these will be set in a shared config file instead.
-func DefaultCentralizedConfigs() *Config {
+func DefaultCentralizedConfigs(cephVersion version.CephVersion) *Config {
 	c := NewConfig()
 	c.Section("global").
+		Set("mon allow pool delete", "true")
+
+	// Everything before Nautilus 14.2.1
```

Comment:
I think it would be good to explain why `14.2.1+` doesn't use these flags. Also, I think it would be good to note somehow that this condition and the condition in `defaultFlagConfigs` are part of the same concept so that it's clear they can both be altered at once when Rook eventually drops support for Nautilus. I think mentioning `default-log-to-file` and `default-mon-cluster-log-to-file` flags here would be a sufficient to explain the connection.